procedures: rewrite and expand the "Backup and disaster recovery" sections of the admin guide (#2121)
codenvy/che-docs: modules/administration-guide/partials/con_backup-and-recovery.adoc

[id="backup-and-recovery_{context}"]
= Backup and recovery

Backing up {prod-short} involves a combination of the following processes that back up respective data:

> * Backing up and restoring the {prod-short} instance by using {prod-short} Operator and a xref:supported-restic-compatible-backup-servers.adoc[configured backup server]. You can back up the {prod-short} instance using custom resources or {prod-cli} to an external backup server that you have configured: xref:backups-of-che-instances-to-an-sftp-backup-server.adoc[SFTP], xref:backups-of-che-instances-to-amazon-s3.adoc[Amazon S3], or xref:backups-of-che-instances-to-a-rest-backup-server.adoc[REST]. You can restore the {prod-short} instance using xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-the-checlusterrestore-object_{context}[custom resources] or xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-prod-cli_{context}[{prod-cli}]. For testing this process in a non-production environment, {prod-short} offers the xref:backups-of-che-instances-to-the-internal-backup-server.adoc[internal backup server].

Reviewer: 1. I recommend to have each sentence in a separate row. This gives some advantages:
  - It is easier to write, no single very long line
  - it is easier to review and I would be able to comment on a specific sentence instead of whole paragraph
  - it is easier for you to find the place which reviewer's comment mentions
  - it is easier to read the diff
2. I think that the page should mention _only_ about overall steps to be done. Information about existing backup servers and the process details should go on appropriate page that describes the flow.
3. It is not clear which link I should click to go to the backup or restore flow... Please make a single link to a page that describe backup/restore flows instead of huge paragraph about everything.

Author: The links make the .adoc version appear about twice as long as the rendered text. Here's the latest version of the rendered text:

![image](https://user-images.githubusercontent.com/78564304/146098871-e086033e-e8e3-4d80-b166-9aea1255728b.png)

The second reason why I put it all into one paragraph is because it's one of three bullet points.

Reviewer: Ok to me

> * Backing up and restoring persisting data about the state of {prod-short} by using xref:backups-of-postgresql.adoc[backups of the PostgreSQL database].

Reviewer: This is actual only for the case when _external_ Postgres is used. I'd move it to the end of the page as a note if external Postgres is used admin have to backup database too.

Author: Moved to the end, but still keeping it as it is since it's an existing functionality.

> * Backing up and restoring the content of users' workspaces by using xref:backups-of-persistent-volumes.adoc[backups of persistent volumes].

Reviewer: I'd add a note that this step might be omitted if all users commit and push their changes. Word `content` is not completely clear here. I think we should stress that we are talking about "source code that is stored inside user's workspaces".

Author: Done. Thank you!

[id="backup-and-recovery_{context}"]
= Backup and recovery

> Backing up {prod-short} involves a combination of the following processes that back up respective data:

Reviewer 2: ```suggestion
Backing up {prod-short} involves a combination of the following processes that back up respective data:
```
It's not clear to me here what `respective` means in this context.

Author: Thanks. I meant `different`.

> * Backing up and restoring the {prod-short} instance by using {prod-short} Operator and a xref:supported-restic-compatible-backup-servers.adoc[configured backup server]. You can back up the {prod-short} instance using custom resources or {prod-cli} to a backup server that you have configured: xref:backups-of-che-instances-to-an-sftp-backup-server.adoc[SFTP], xref:backups-of-che-instances-to-amazon-s3.adoc[Amazon S3] (or S3 API compatible storage), or xref:backups-of-che-instances-to-a-rest-backup-server.adoc[REST]. You can restore the {prod-short} instance using xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-the-checlusterrestore-object_{context}[custom resources] or xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-prod-cli_{context}[{prod-cli}].

Reviewer 2: ```suggestion
* Use the {prod-short} Operator and a xref:supported-restic-compatible-backup-servers.adoc[configured backup server] to back up and restore the {prod-short} instance. You can back up the {prod-short} instance using custom resources or {prod-cli} to a backup server that you have configured: xref:backups-of-che-instances-to-an-sftp-backup-server.adoc[SFTP], xref:backups-of-che-instances-to-amazon-s3.adoc[Amazon S3] (or S3 API compatible storage), or xref:backups-of-che-instances-to-a-rest-backup-server.adoc[REST]. You can restore the {prod-short} instance using xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-the-checlusterrestore-object_{context}[custom resources] or xref:restoring-a-che-instance-from-a-backup.adoc#restoring-a-che-instance-from-a-backup-using-prod-cli_{context}[{prod-cli}].
```
Edit to avoid the sentence fragment and make more active.

+

> NOTE: For testing this process, {prod-short} offers the xref:backups-of-che-instances-to-the-internal-backup-server.adoc[internal backup server].

Reviewer 2: ```suggestion
NOTE: To test this process, use the xref:backups-of-che-instances-to-the-internal-backup-server.adoc[internal backup server].
```
- edit for idiomatic usage and to make more user-centric.
- also, is this the correct formatting for a Note admonition?

Author: @bburt-rh. Fixed. Thank you.
This is the formatting that supports single-paragraph admonitions:
https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/

> * Backing up and restoring the source code stored in users' workspaces by using xref:backups-of-persistent-volumes.adoc[backups of persistent volumes].

Reviewer 2: ```suggestion
* Use  xref:backups-of-persistent-volumes.adoc[backups of persistent volumes] to back up and restore the source code stored in users' workspaces.
```
Avoid the sentence fragment and make more user-centric with a direct imperative.

+

> TIP: Whether or not backups of persistent volumes are implemented, users can be advised to commit and push their changes to avoid losing their work.

Reviewer 2: ```suggestion
TIP: Whether or not you implement backups of persistent volumes, advise users to commit and push their changes to avoid losing their work.
```
- Avoid passive voice. Be authoritative.
- Is this the correct formatting for a Tip admonition?

Author: Thanks. Fixed.

> * Use xref:backups-of-postgresql.adoc[backups of the PostgreSQL database] to back up and restore persisting data about the state of {prod-short}.

Reviewer: This is actual only for external PostgreSQL. In case of internal one, standard backup/restore takes care of it.

Author: @mmorhun Thank you. I've just added `external` to the `xref`.

I have already reused (from the published docs) the following in another file in this PR:
```
= External PostgreSQL setup

By default, the {prod-short} Operator creates and manages deployment of the internal PostgreSQL database. However, the {prod-short} Operator does not support full lifecycle capabilities such as backups and recovery.

A business-critical setup must use an external PostgreSQL database ...
```

> * Backing up and restoring persisting data about the state of {prod-short} by using xref:backups-of-postgresql.adoc[backups of the PostgreSQL database].

Reviewer 2: ```suggestion
* Use xref:backups-of-postgresql.adoc[backups of the PostgreSQL database] to back up and restore persistent data about the state of {prod-short}.
```
Avoid the sentence fragment and make more active.

Author: Great! Thank you! Fixed.
